noir-r1cs/src/noir_proof_scheme.rs

```diff
@@ -94,18 +100,54 @@ impl NoirProofScheme {
         (self.r1cs.num_constraints(), self.r1cs.num_witnesses())
     }
 
+    pub fn read_witness(&self, prover_toml: impl AsRef<Path>) -> Result<InputMap> {
+        let (input_map, _expected_return) =
+            read_inputs_from_file(prover_toml.as_ref(), self.witness_generator.abi())?;
+
+        Ok(input_map)
+    }
+
+    pub fn generate_witness(&self, input_map: &InputMap) -> Result<WitnessMap<NoirFieldElement>> {
+        let span = span!(Level::INFO, "generate_witness").entered();
```

Comment:
Question, why use this over the `instrument` macro?

Reply:
I think I just copied it other, from what was here before, fixing now